include/psa/crypto_struct.h

```diff
@@ -223,6 +223,29 @@ static inline struct psa_key_derivation_s psa_key_derivation_operation_init(
     return v;
 }
 
+struct psa_key_generation_method_s {
+    /* Future versions may add other fields in this structure. */
+    uint32_t flags;
+    uint8_t data[];
```

Comment:
The length of `data` is the length of the whole method-with-data minus the size of the structure up to the `data` field. In other words, if the length of the whole method-with-data is `method_length` (which [is passed to the API functions](https://github.com/Mbed-TLS/mbedtls/pull/8815/files#r1486530094)), then the length of the data is `method_length - offsetof(psa_key_generation_method_t, data)`.

Note that depending on the presence of implementation-specific or to-be-added fields, and depending on the platform, the structure may have padding. So it's possible that `offsetof(psa_key_generation_method_t, data) < sizeof(psa_key_generation_method_t)`.